docs/src/main/tut/invariant.md

```diff
@@ -0,0 +1,78 @@
+---
+layout: default
+title:  "Invariant"
+section: "typeclasses"
+source: "https://github.com/non/cats/blob/master/core/src/main/scala/cats/functor/Invariant.scala"
+scaladoc: "#cats.functor.Invariant"
+---
+# Invariant
+
+The `Invariant` typeclass is for `Functor`s that define an `imap`
+function with the following type:
+
+```scala
+def imap[A, B](fa: F[A])(f: A => B)(g: B => A): F[B]
+```
+
+Every covariant (as well as contravariant) functor gives rise to an invariant
+functor, by ignorig the `f` (or in case of contravariance, `g`) function.
+
+Examples for instances of `Invariant` are `Semigroup` and `Monoid`, in
+the following we will explain why this is the case using `Semigroup`, the
+reasoning for `Monoid` is analogous.
+
+## Invariant instance for Semigroup
+
+Pretend that we have a `Semigroup[Long]` representing a standard UNIX
+timestamp.  Let's say that we want to create a `Semigroup[Date]`, by
+*reusing* `Semigroup[Long]`.
+
+### Semigroup does not form a covariant functor
+
+If `Semigroup` had an instance for the standard covariant `Functor`
+typeclass, we could use `map` to apply a function `longtoDate`:
+
+```scala
+def longToDate: Long => Date = new Date(_)
```

Comment:
This comment applies to some other places below. You can use `sbt docs/tut` to make sure the docs compile after these changes.